COSC481W-2024Fall/MLite
# Scaffold Dataset Model  and File Upload Validation

Issue #4 
**This PR is ready for review**

Features
- Users can now manage datasets with a CRUD interface.
- Scaffolded a new Dataset model with the following attributes:
  - `name`: string
  - `description`: text
  - `file`: ActiveStorage
  - `size`: integer
  - `columns`: integer
  - `n_rows`: integer
  - `dataset_type`: string
  - `user_id`: integer
  - `metrics`: JSON
- Implemented basic validations for file upload limits and ensuring the file is in CSV format.
 

Code
- Added a scaffold for the Dataset model.
- Updated the datasets controller and views for basic CRUD functionality.
- Validations for file upload limits and CSV format are in place.
- Configured ActiveStorage for handling file uploads.

Brief Code Overview
`db/migrate/:` Migration file to create the datasets table in the database.
`dataset.rb:` Model file for the Dataset table, including validations for CSV files and file size.
`views/datasets/:` Views for managing datasets (e.g., index, new, show, edit).
`controllers/datasets_controller.rb:` Controller logic for handling dataset CRUD operations.
`config/storage.yml:` Configurations for ActiveStorage setup.

Testing

1. Pull this branch.
2. Run `rails db:migrate.`
3. Restart the server with `rails s` (the server needs to be restarted after the migration).
4. Navigate to `localhost:3000/datasets/new` to upload a dataset.
5. Only CSV files are allowed.
6. After uploading, go to `localhost:3000/datasets` to view the list of all uploaded datasets.
7. Test dataset deletion by clicking the delete button next to a dataset.

What to look out for:
- Ensure that the CRUD functionality works as expected.
- Verify that only CSV files are allowed for uploads.
- Dataset deletion.

Note on this PR
This is primarily a feature addition for dataset management, focusing on scaffolding and basic CRUD operations.
